dot/peerset/peerset_test.go

```diff
@@ -78,9 +79,15 @@ func TestAddReservedPeers(t *testing.T) {
 
 	reservedPeers := peer.IDSlice{reservedPeer, reservedPeer2}
 
+	var waitGroup sync.WaitGroup
+
 	for _, peerID := range reservedPeers {
-		handler.AddReservedPeer(testSetID, peerID)
-		time.Sleep(time.Millisecond * 100)
+		waitGroup.Add(1)
+		go func() {
+			handler.AddReservedPeer(testSetID, peerID)
+			waitGroup.Done()
+		}()
+		waitGroup.Wait()
```

Comment:
This doesn't do anything (same as just `handler.AddReservedPeer(testSetID, peerID)`)

I think we want to wait for a fire-and-forget goroutine launched within `AddReservedPeer`, so that would require a refactor of the production code of the method `AddReservedPeer`. Just a wild guess though, I haven't really digged into that method either.

Same comment for the other changes.

Parallel programming is hard too, so take it easy 😉